Source/Config/Mixins/Commodities.lua

```diff
@@ -0,0 +1,53 @@
+AuctionatorConfigCommoditiesFrameMixin = CreateFromMixins(AuctionatorPanelConfigMixin)
+
+function AuctionatorConfigCommoditiesFrameMixin:OnLoad()
+  Auctionator.Debug.Message("AuctionatorConfigCommoditiesFrameMixin:OnLoad()")
+
+  self.name = "   Commodities"
+  self.parent = "Selling"
+
+  self:SetupPanel()
+
+  self.CommoditySalesPreference:SetOnChange(function(selectedValue)
+    self:OnSalesPreferenceChange(selectedValue)
+  end)
+end
+
+function AuctionatorConfigCommoditiesFrameMixin:OnShow()
+  self.currentCommodityDuration = Auctionator.Config.Get(Auctionator.Config.Options.COMMODITY_AUCTION_DURATION)
+  self.currentCommiditySalesPreference = Auctionator.Config.Get(Auctionator.Config.Options.COMMODITY_AUCTION_SALES_PREFERENCE)
+
+  self.CommodityDurationGroup:SetSelectedValue(self.currentCommodityDuration)
+  self.CommoditySalesPreference:SetSelectedValue(self.currentCommiditySalesPreference)
+
+  self:OnSalesPreferenceChange(self.currentCommiditySalesPreference)
+
+  self.CommodityUndercutPercentage:SetNumber(Auctionator.Config.Get(Auctionator.Config.Options.COMMODITY_UNDERCUT_PERCENTAGE))
+  self.CommodityUndercutValue:SetAmount(Auctionator.Config.Get(Auctionator.Config.Options.COMMODITY_UNDERCUT_STATIC_VALUE))
+end
+
+function AuctionatorConfigCommoditiesFrameMixin:OnSalesPreferenceChange(selectedValue)
+  self.currentCommiditySalesPreference = selectedValue
+
+  if self.currentCommiditySalesPreference == Auctionator.Config.SalesTypes.PERCENTAGE then
+    self.CommodityUndercutPercentage:Show()
+    self.CommodityUndercutValue:Hide()
+  else
+    self.CommodityUndercutValue:Show()
+    self.CommodityUndercutPercentage:Hide()
+  end
+end
+
+function AuctionatorConfigCommoditiesFrameMixin:Save()
+  Auctionator.Debug.Message("AuctionatorConfigCommoditiesFrameMixin:Save()")
+
+  Auctionator.Config.Set(Auctionator.Config.Options.COMMODITY_AUCTION_DURATION, self.CommodityDurationGroup:GetValue())
+
+  Auctionator.Config.Set(Auctionator.Config.Options.COMMODITY_AUCTION_SALES_PREFERENCE, self.CommoditySalesPreference:GetValue())
+  Auctionator.Config.Set(Auctionator.Config.Options.COMMODITY_UNDERCUT_PERCENTAGE, tonumber(self.CommodityUndercutPercentage:GetNumber()))
```

Comment:
No validation of the % entered. Found I could put in 999%.

Reply:
Validate the input rather that just changing the characters of the text box (otherwise its vulnerable to UI changes outside our control getting bad values through)